enews-node/newsFetcher.js

```diff
@@ -0,0 +1,72 @@
+module.exports = NewsFetcher;
+
+function NewsFetcher(token) {
+  const GENERAL_CHANNEL_ID = 'C029J9QTH'
+  const HISTORY_ENDPOINT = 'https://slack.com/api/channels.history'
+  const MESSAGE_COUNT = 1000;
+
+  var httpClient = require('request');
+
+  this.getEnews = function getEnews(oldest, latest, callback) {
+    var allEnews = [];
+    var handler = function(enews, hasMore, reducedLatest) {
+      allEnews = allEnews.concat(enews);
+      if (hasMore) {
+        getSlackHistory(createHistoryRequest(oldest, reducedLatest), handler);
+      } else {
+        callback(allEnews);
+      }
+    }
+    getSlackHistory(createHistoryRequest(oldest, latest), handler);
+  }
+
+  function getSlackHistory(request, callback) {
+    httpClient.get(request, function(error, response, body) {
+      var jsonBody = JSON.parse(body);
+      var messages = jsonBody.messages;
+      var enews = parseMessages(messages);
+      var reducedLatest = messages[messages.length - 1].ts;
+      callback(enews, jsonBody.has_more, reducedLatest);
+    });
+  }
+
+  function parseMessages(messages) {
+    var enews = [];
+    messages.forEach(function(message) {
+      if (isEnewsMessage(message)) {
+        enews.push(message.text);
+      }
+    });
+    return enews;
+  }
+
+  function isEnewsMessage(message) {
+    return isValidMessage(message) && isEnews(message.text)
+  }
+
+  function isEnews(messageText) {
+    return contains(messageText, '#enews') || contains(messageText, '#eNews');
```

Comment:
you might want to add a check for the text `http` because some people talk about the `#eNews` tag without actually posting any url! ( "can you post the #eNews you said in standup") for example.

Also you might want to `lowerCase` the `messageText` and just check for `#enews` as some peoples fat fingers have used the hashtag `#ENews` which isn't caught here